src/Engine/Component/Shape.hpp

```diff
@@ -2,26 +2,43 @@
 
 #include "SuperComponent.hpp"
 
-class btCollisionShape;
-class Entity;
+class PhysicsManager;
+
+namespace Physics {
+    class Shape;
+}
 
 namespace Component {
     /// Physical shape that can be used in conjunction with rigid bodies or to
     /// act as a trigger volume.
     class Shape : public SuperComponent {
-        friend class PhysicsManager;
+        friend class ::PhysicsManager;
 
         public:
             /// Create new shape.
-            Shape();
+            Shape() = default;
             
             /// Save the component.
             /**
              * @return JSON value to be stored on disk.
              */
             Json::Value Save() const override;
-            
+
+            /// Get the shape this component represents.
+            /**
+             * @return The %Shape.
+             */
+            ::Physics::Shape& GetShape();
+            const ::Physics::Shape& GetShape() const;
```

Comment:
I think this needs a separate doxygen comment. It won't just use the above methods comments.